operator/pkg/model/ingestion/ingress.go

```diff
@@ -0,0 +1,197 @@
+		Kind:      "Ingress",
+	}
+
+	if ing.Spec.DefaultBackend != nil {
+		// There's a default backend set up
+
+		// get the details for the default backend
+
+		backend := model.Backend{}
+		backend.Name = ing.Spec.DefaultBackend.Service.Name
+		backend.Namespace = ing.Namespace
+
+		backend.Port = &model.BackendPort{}
+
+		if ing.Spec.DefaultBackend.Service.Port.Name != "" {
+			backend.Port.Name = ing.Spec.DefaultBackend.Service.Port.Name
+		}
+
+		if ing.Spec.DefaultBackend.Service.Port.Number != 0 {
+			backend.Port.Port = uint32(ing.Spec.DefaultBackend.Service.Port.Number)
+		}
+
+		l := model.HTTPListener{
+			Hostname: "*",
+			Routes: []model.HTTPRoute{
+				{
+					Backends: []model.Backend{
+						backend,
+					},
+				}},
+			Port: 80,
+		}
+
+		l.Sources = model.AddSource(l.Sources, sourceResource)
+
+		insecureListenerMap["*"] = l
+	}
+
+	// Now, we range across the rules, adding them in as listeners.
+	for _, rule := range ing.Spec.Rules {
+
+		host := "*"
+
+		if rule.Host != "" {
+			host = rule.Host
+		}
+
+		l, ok := insecureListenerMap[host]
+		l.Port = 80
+		l.Sources = model.AddSource(l.Sources, sourceResource)
+		if !ok {
+			l.Name = fmt.Sprintf("ing-%s-%s-%s", ing.Name, ing.Namespace, host)
+		}
+
+		l.Hostname = host
+		for _, path := range rule.HTTP.Paths {
+
+			route := model.HTTPRoute{}
+
+			switch *path.PathType {
+			case slim_networkingv1.PathTypeExact:
+				route.PathMatch.Exact = path.Path
+			case slim_networkingv1.PathTypePrefix:
+				route.PathMatch.Prefix = path.Path
+			case slim_networkingv1.PathTypeImplementationSpecific:
+				route.PathMatch.Regex = path.Path
+			default:
+				// TODO(youngnick): Figure out what to do in this case.
+				continue
+			}
+
+			backend := model.Backend{
+				Name:      path.Backend.Service.Name,
+				Namespace: ing.Namespace,
+			}
+			if path.Backend.Service != nil {
+				backend.Port = &model.BackendPort{}
+				if path.Backend.Service.Port.Name != "" {
+					backend.Port.Name = path.Backend.Service.Port.Name
+				}
+				if path.Backend.Service.Port.Number != 0 {
+					backend.Port.Port = uint32(path.Backend.Service.Port.Number)
+				}
+			}
+			route.Backends = append(route.Backends, backend)
+			l.Routes = append(l.Routes, route)
+		}
+
+		insecureListenerMap[host] = l
+	}
+
+	secureListenerMap := make(map[string]model.HTTPListener)
+
+	// First, we check for TLS config, and set them up with Listeners to return.
+	for _, tlsConfig := range ing.Spec.TLS {
+
+		for _, host := range tlsConfig.Hosts {
+
+			var l model.HTTPListener
+
+			l, ok := secureListenerMap[host]
+			if !ok {
+				l = insecureListenerMap[host]
+			}
+
+			l.Sources = model.AddSource(l.Sources, sourceResource)
+			if tlsConfig.SecretName != "" {
+				l.TLS = &model.TLSSecret{
+					Name: tlsConfig.SecretName,
+					// Secret has to be in the same namespace as the Ingress.
+					Namespace: ing.Namespace,
+				}
+
+			}
+			l.Port = 443
+			l.Hostname = host
+			secureListenerMap[host] = l
```

Comment:
It seems like this will always add to the secureListenerMap. Does this mean that the same host intentionally will end up in both `insecureListenerMap` and `secureListenerMap`? What does that mean for allowing connectivity both encrypted/not on ports 80,443?

The comments above the function didn't really specify expectations like this for the output format. Maybe it's obvious if you deal with these resources a lot, but it wasn't entirely obvious to me at first read.

Reply:
good question, the Ingress conformance tests didn't actually enforce if a particular rule is https only i.e. it's fine to have both secure and insure listeners.

In later commit, you might see some changes related to enforceHttps, in which will return 301 http error with details about HTTPs.

Reply:
Ah right, I thought the 301 http error stuff might be the point behind this, but I didn't stick around the PR long enough to figure out how that fits in :sweat_smile: